artiq/firmware/libboard_misoc/i2c.rs

```diff
@@ -188,13 +188,16 @@ mod imp {
         Ok(data)
     }
 
-    pub fn pca9548_select(busno: u8, address: u8, channels: u8) -> Result<(), &'static str> {
+    pub fn switch_select(busno: u8, address: u8, mask: u8) -> Result<(), &'static str> {
+        // address in 7-bit form
+        // mask in format of 1 << channel (or 0 for disabling output)
+        // PCA9548 support only for now
         start(busno)?;
         if !write(busno, address << 1)? {
-            return Err("PCA9548 failed to ack write address")
+            return Err("I2C Switch failed to ack write address")
```

Comment:
We know it's a PCA9548 at this point. This error message does not need to be edited.